src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -111,6 +111,11 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   mCommonAngleActions = QMap<QAction *, int>();
   QList< QPair< int, QString > > commonAngles;
   commonAngles << QPair<int, QString>( 0, tr( "Do not snap to common angles" ) );
+  commonAngles << QPair<int, QString>( 5, tr( "Snap to 5° angles" ) );
+  commonAngles << QPair<int, QString>( 10, tr( "Snap to 10° angles" ) );
+  commonAngles << QPair<int, QString>( 15, tr( "Snap to 15° angles" ) );
+  commonAngles << QPair<int, QString>( 18, tr( "Snap to 18° angles" ) );
+  commonAngles << QPair<int, QString>( 23, tr( "Snap to 23° angles" ) );
```

Comment:
shall read 24°? (360/23 is not an integer)

Reply:
ah from your printscreen it should be 22.5.

I am a bit reluctant at adding so much. It is still easily achievable through the widget by entering the angle (and eventually locking it).